### ISSUE TITLE
Is there any update in the last 4 Years?

### ISSUE THREAD

**Issue by LuckyLuzz** [sequoia-hope/ReprapHeatedBedGenerator#2]
Hey,
thanks for your awesome Project. Id like to as you if there are any updates made in 4 Years? 

Would it be possible to add support for bore holes?

**Comment by sequoia-hope** [comment]
Hello!

No updates in four years. What do you mean by Bore Holes?

**Comment by LuckyLuzz** [comment]
Hey,
thanks for you answer. If you have a look at the http://reprap.org/wiki/PCB_Heatbed       you will see that thes beds have holes for screws or thermistors. So I meant to implement a function that first generate holes like hole(x,y,diameter) and then leads the traces around the holes. Would that be possile?

Also Processing 2.0  is not to download anymore. The next version 2.2.1 is not working.  
https://processing.org/download/?processing

```
board.xml does not exist or could not be read
java.lang.NullPointerException
	at processing.data.XML$1.read(XML.java:190)
        .........
```

**Comment by sequoia-hope** [comment]
I recommend just adding holes after the fact. But if you want to make it
automated this is possible. The Eagle files are just XML. All this program
does is add text to the XML document that defines the board. If you make a
blank board and add one hole in Eagle then save it, you can open the brd
file in a text editor to see how a hole is defined in XML. Does that make
sense?

**Comment by LuckyLuzz** [comment]
That does make sense to me. But since I am a noob in Java that will take a while... 
What I am going to do then is first: Update to 2.2.1 or higher with a remix of your skript as example found here. http://www.thingiverse.com/thing:456670    Second: Find out how the brd files are working and implement a function for holes. 

What would be you guess for the hole function?

**Comment by LuckyLuzz** [comment]
Got it running on 2.2.1. Had to copy the board.xml to in the same folder as the pde is. Now Second :)

**Comment by sequoia-hope** [comment]
And just to be clear... If you're just making one board you can easily add
the holes manually using Eagle. You only need to mess with the code if you
want the program to be able to automatically add holes.

Have you tried adding holes using Eagle after the board is generated?

**Comment by LuckyLuzz** [comment]
yeah I know, I wanna do that so DIY 3D Printer folks can customize their beds to their need as close as possible.  I guess that was your intend to create that Tool, too. And it's awesome, just a bit improvement left :)
No I dont, I was just about to test Processing 3.2.3, looks like it works also.

Another Idea that I discussed in a Forum was to create a bed who outer sides will heat more than the middle part, that would compensate heat los.

**Comment by boelle** [comment]
Hi

just came across this,

a nice feature would be zones so that you could have a circle in the middle and 4 zones arround it.

more or less what the mk42 bed does to prevent cold corners
